### PR TITLE
PLT-661 Changed all goroutine functions to use '...AndForget' as the naming convention

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Pretty self-explanatory, this renames all functions that only consist of a goroutine to end with 'AndForget'.  This includes some functions that previously had no reference to any convention (e.g. didn't start with 'FireAndForget...') so please ping me if any of those we're intended to be left out.

### CODE REVIEW

**Comment by hmhealey** [comment]
Looks good

File: api/web_conn.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ type WebConn struct {
 	ChannelAccessCache map[string]bool
 }
 
-func NewWebConn(ws *websocket.Conn, teamId string, userId string, sessionId string) *WebConn {
+func NewWebConnAndForget(ws *websocket.Conn, teamId string, userId string, sessionId string) *WebConn {
```
Comment (jwilander) [inline api/web_conn.go]: This one should still be `NewWebConn` since there's a synchronous return

File: api/web_hub.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func (h *Hub) Stop() {
 	h.stop <- "all"
 }
 
-func (h *Hub) Start() {
+func (h *Hub) StartAndForget() {
```
Comment (jwilander) [inline api/web_hub.go]: Can we also leave all the `Start` functions the way they were? Sorry, I should have mentioned those as a special case before

File: api/web_conn.go (hunk 1 above)
Comment (rgarmsen2295) [inline api/web_conn.go]: Yup I'll change it back, thanks!

File: api/web_hub.go (hunk 2 above)
Comment (rgarmsen2295) [inline api/web_hub.go]: Yup can do, that was one of the ones I wasn't quite sure about but since `PublishAndForget` was above I went with it.

**Comment by coreyhulen** [comment]
I agrees with jwilander's comments, othewise +1

**Comment by rgarmsen2295** [comment]
Ready for re-review when possible @jwilander

**Pull request merged**